glom/core.py

```diff
@@ -1821,14 +1864,24 @@ def glom(target, spec, **kwargs):
     scope[ROOT] = scope
     scope[T] = target
     scope.update(kwargs.pop('scope', {}))
+    err = None
     if kwargs:
         raise TypeError('unexpected keyword args: %r' % sorted(kwargs.keys()))
     try:
-        ret = _glom(target, spec, scope)
-    except skip_exc:
-        if default is _MISSING:
-            raise
-        ret = default
+        try:
+            ret = _glom(target, spec, scope)
+        except skip_exc:
+            if default is _MISSING:
+                raise
+            ret = default
+    except Exception as e:
+        if isinstance(e, GlomError):
+            err = e
+        else:
+            err = GlomError.wrap(e)
+        err._finalize(scope[ERROR_SCOPE])
+    if err:
+        raise err
```

Comment:
and we've got to play this game because python 3 has some weird scoping issue, right?

Reply:
without this, the whole stack trace is displayed even though it is a different error somehow